core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -201,19 +201,12 @@ export class OptimizedSsrEngine {
 
     const renderingKey = this.getRenderingKey(request);
 
-    /**
```

Comment:
Let's add a test that covers the following scenario:

The concurrency is e.g. 1, and reuse rendering is on. After the first request trigger the render, there's another request for the same URL.
The expected behavior is the second request should wait, but currently it's immediately falling back to CSR.